### ISSUE TITLE
Cloud search should weight fields differently

### ISSUE THREAD

**Issue by johntseng** [sillsdev/webonary#406]
After #405 and we have all the fields populated, we should weight the search fields by something like 
https://github.com/sillsdev/webonary/blob/4bde946f8d67935025e259ad1a3be96f193b2907/wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Pathway_Xhtml_Import.php#L480

**Comment by vernastutzman** [comment]
The Search results should be returned in a relevant order by fields.

Here is the order for searching for something in the vernacular language; headword in main entry,  headword in minor entry, headword in subentry, any cross references, including lexical relations, example sentences, then everything else.

Here is the order for searching for something in the analysis languages. Definition, translation of example sentences, notes, then everything else.

**Comment by vernastutzman** [comment]
See Issue #397

**Comment by johntseng** [comment]
Duplicate of #397

**Comment by vernastutzman** [comment]
There is still a difference in the search order results depending on whether the match whole words is checked or not. The correct order occurs when you Match whole words.
https://www.webonary.work/test-stutzman7?s=lion&search=Search&key=fr&pos=&semantic_domain=&search_options_set=1
![image](https://user-images.githubusercontent.com/53490028/233668803-5923beac-3352-44cd-9d45-d325d6945603.png)
![image](https://user-images.githubusercontent.com/53490028/233668951-9c063cff-4cae-4177-b2e4-0695a96068f9.png)

**Comment by vernastutzman** [comment]
Reference prior example.
